### PR TITLE
(🎁) Add 3.12-dev to CI

### PR DESCRIPTION
**Pull request by KotlinIsland**
Adding a fail-able 3.12-dev job to the CI

### CODE REVIEW

**Comment by KotlinIsland** [comment]
Very strange, pytest seems to timeout github right near the end, I'll continue investigating when I have time.

**Comment by JelleZijlstra** [comment]
Maybe make it skip the lxml install; it's probably not worth it to try to build from source, and we can still get useful data from tests that don't rely on XML reports.

**Comment by hauntsaninja** [comment]
Yup, do the opposite of #14134

**Comment by hauntsaninja** [comment]
Hmm, 40 minutes and counting. Doesn't bode well

**Comment by KotlinIsland** [comment]
Yeah, there is some issue with the tests right at the end, I'm temporarily increasing pytest verbosity and rerunning.

**Comment by KotlinIsland** [comment]
@JelleZijlstra @hauntsaninja If you see this before it times out (6 hours), could you cancel the current run so I can analyze the logs?

**Comment by JelleZijlstra** [comment]
done

**Comment by KotlinIsland** [comment]
It's `mypyc/test/test_run.py::TestRun::run-functions.test::testDecorators1` that it's hanging on.

File: mypyc/test/test_run.py [hunk 1]
```diff
@@ -312,6 +312,8 @@ def run_case_step(self, testcase: DataDrivenTestCase, incremental_step: int) ->
             stderr=subprocess.STDOUT,
             env=env,
         )
+        if sys.version_info >= (3, 12):
+            proc.wait(timeout=30)
```
[the remark this answers is not in the document]
Comment (KotlinIsland) [reply mypyc/test/test_run.py:316]: It doesn't attempt to fix the test at all, it just stops the bad test from it running forever.

I don't really know what's going on here. And I can't figure out a (sane) way of retrieving the output from the process without hanging.

[the remark this answers is not in the document]
Comment (hauntsaninja) [reply mypyc/test/test_run.py:316]: Oh I see, I got confused by the green status check, I do see the TimoutExpired in the log

Comment (hauntsaninja) [inline mypyc/test/test_run.py:316]:
```suggestion
        if sys.version_info >= (3, 12):
            # TODO: testDecorators1 hangs on 3.12
            proc.wait(timeout=30)
```

**Review by hauntsaninja — APPROVED** [review]
Okay, this seems fine to me. Let me know if you're interested in fixing any of the failing tests and need help

**Comment by KotlinIsland** [comment]
> Okay, this seems fine to me. Let me know if you're interested in fixing any of the failing tests and need help

If I have time I can look into it, is this good to merge now?

**Pull request merged**